### PR TITLE
fix: added new end point for aws get token flow

### PR DESCRIPTION
**Pull request by vigilantee**
Removing the dependency for passing the Client Id for aws cognito flow for getiing the token.

### CODE REVIEW

File: frappe/integrations/oauth2_logins.py [hunk 1]
```diff
@@ -47,6 +49,19 @@ def get_login_url(redirect_to=None, social_login_key_name="bloomstack"):
 		redirect_to = redirect_to.replace("&amp;", "&")
 	return get_oauth2_authorize_url(social_login_key_name, redirect_to)
 
+# A method to retrieve the client id from client name and call get_token for Oauth Flow
```
Comment (hrwX) [inline frappe/integrations/oauth2_logins.py]: Add this as docstring

**Review by hrwX — CHANGES_REQUESTED** [review]

File: frappe/integrations/oauth2_logins.py (hunk 1 above)
Comment (vigilantee) [reply frappe/integrations/oauth2_logins.py]: ok sure.

Comment (vigilantee) [reply frappe/integrations/oauth2_logins.py]: done.

**Review by sahil28297 — DISMISSED** [review]

**Review by neilLasrado — APPROVED** [review]

File: frappe/integrations/oauth2_logins.py [hunk 2]
```diff
@@ -47,6 +49,27 @@ def get_login_url(redirect_to=None, social_login_key_name="bloomstack"):
...
+	"""
+	:API endpoint to retrieve the client id from client name and call get_token for Oauth Flow
+	
+	:param grant_type: one of authorization_code, refresh_token, password.
+	:param client_name: Client that you need token for.
+	:param redirect_uri: Will be redirected to post recieving the token.
+	:param code: one time token generated by aws.
+	:returns: Bearer token.
+	"""
+    clients = frappe.db.get_list('OAuth Client', filters={'app_name': client_name},fields=['client_id'])
```
Comment (neilLasrado) [inline frappe/integrations/oauth2_logins.py]: Won't the get_list fail if the user is not logged in? You should use get_all instead

**Review by neilLasrado — CHANGES_REQUESTED** [review]

File: frappe/integrations/oauth2_logins.py [hunk 3]
```diff
@@ -47,6 +49,27 @@ def get_login_url(redirect_to=None, social_login_key_name="bloomstack"):
...
+    client_id = clients[0].client_id if clients else None
+    if not client_id:
+        return frappe.AuthenticationError
+    newForm = frappe.request.form.to_dict()
```
Comment (vishdha) [inline frappe/integrations/oauth2_logins.py:67]: @vigilantee  frappe.request dont have any form function.
you may try this
frappe.form_dict

File: frappe/integrations/oauth2_logins.py (hunk 2 above)
Comment (vigilantee) [reply frappe/integrations/oauth2_logins.py]: This api would be called post user authentication by aws.

**Comment by hrwX** [comment]
@dti-deploy create instance

**Comment by hrwX** [comment]
@dti-deploy create instance

File: frappe/integrations/oauth2_logins.py [hunk 4]
```diff
@@ -40,13 +42,34 @@ def get_login_url(redirect_to=None, social_login_key_name="bloomstack"):
...
+	:returns: Bearer token.
+	"""
+	clients = frappe.db.get_all('OAuth Client', filters={'app_name': client_name},fields=['client_id'])
+	client_id = clients[0].client_id if clients else None
+	if not client_id:
+		return frappe.AuthenticationError
+	newForm = frappe.request.form.to_dict()
+	newForm["client_id"] = client_id
+	newForm = ImmutableMultiDict(newForm)
+	frappe.request.form = newForm
```
Comment (hrwX) [inline frappe/integrations/oauth2_logins.py]:
```suggestion
	form_dict = frappe.request.form.to_dict()
	form_dict["client_id"] = client_id
	form_dict = ImmutableMultiDict(form_dict)
	frappe.request.form = form_dict
```

**Review by hrwX — CHANGES_REQUESTED** [review]

**Review by hrwX — APPROVED** [review]

**Review by sahil28297 — APPROVED** [review]

**Pull request merged**